templates/virt-install-script.sh.j2

```diff
@@ -1,29 +1,30 @@
-#!/bin/bash
-
-if [ ! -f /etc/libvirt/qemu/{{ inventory_hostname }}.xml ]; then
-
 virt-install \
   --name={{ inventory_hostname }} \
-  --os-variant=rhel6 \
-{% if virt_install_cpu_model %}
-  --cpu={{ virt_install_cpu_model }} \
-{% else %}
-  --cpu=host \
-{% endif %}
   --vcpus={{ vcpus }} \
   --ram={{ ram }} \
 {% for disk in disks %}
-  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format=raw \
+  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format={{ disk.format | default('raw') }} \
 {% endfor %}
 {% for bridge in bridges %}
   --network bridge={{ bridge }},model=virtio \
 {% endfor %}
   --connect=qemu:///system \
-  --location={{ install_url }} \
-  --graphics=vnc,keymap="fi" \
-  --noautoconsole \
   --autostart \
+  --noautoconsole \
+  --events on_poweroff=preserve \
```

Comment:
**Without** `on_poweroff=preserve` setting, it happens so that when a VM is rebooted the next time with "_shutdown -r now_", it will go to stopped state in virsh rather than restarting. This is designed behaviour of virt-install, as it is originally designed to install VMs and then let them go offline to capture the generated image. It is flag "--no-reboot" for kvm that makes the behaviour of VM not recovering from reboot.
I'm not able to comment on the behaviour with kickstart deployments as I have done only image based deployments.